src/app/utils/researchNowUtils.js

```diff
@@ -15,24 +8,45 @@ const mapSearchScope = {
   date: 'date',
 };
 
-const filterObj = (field, value) => ({ field, value });
-const queryObj = (field, query) => ({ field, query });
+/**
+ *  Given a hash of RC filter params including:
+ *   - `dateAfter`
+ *   - `dateBefore`
+ *   - `language`
+ *
+ *  Returns a hash representing an equivalent `filter` param on DRBAPI
+ */
 
 const mapFilters = (filters = {}) => {
-  const researchNowFilters = [];
+  let researchNowFilters = [];
 
   const years = {};
-  if (filters.dateAfter) years.start = filters.dateAfter;
-  if (filters.dateBefore) years.end = filters.dateBefore;
-  if (years.start || years.end) researchNowFilters.push(filterObj('years', years));
+  if (filters.dateAfter) researchNowFilters.push(`startYear:${filters.dateAfter}`)
+  if (filters.dateBefore) researchNowFilters.push(`endYear:${filters.dateBefore}`)
   if (filters.language) {
-    const languages = filters.language.map(lang => lang.replace('lang:', ''));
-    languages.forEach(language => researchNowFilters.push(filterObj('language', language)));
+    researchNowFilters = researchNowFilters.concat(
+      (Array.isArray(filters.language) ? filters.language : [ filters.language ])
+        .map(lang => lang.replace('lang:', ''))
+        .map(lang => `language:${lang}`)
```

Comment:
Could this be a single map `lang => lang.replace('lang:', 'language:')`?

Reply:
Yeah that's probably safe. Changed.